xarray/tests/test_backends_datatree.py

```diff
@@ -396,15 +396,12 @@ def test_phony_dims_warning(self, tmpdir) -> None:
                 }
 
 
-@pytest.mark.skipif(
-    have_zarr_v3, reason="datatree support for zarr 3 is not implemented yet"
-)
-@requires_zarr
+@requires_zarr_v3
 class TestZarrDatatreeIO:
     engine = "zarr"
 
     def test_to_zarr(self, tmpdir, simple_datatree):
-        filepath = tmpdir / "test.zarr"
+        filepath = str(tmpdir / "test.zarr")
```

Comment:
Did Zarr2 handle this case better? If so, we can probably fix this upstream.

Reply:
See https://github.com/zarr-developers/zarr-python/pull/2911 for a upstream fix.

Reply:
after thinking about this, I agree with the proposed changes here.